"""
Field descriptions extracted from XML files
"""

FIELD_DESCRIPTIONS = {
    "16S_recovery_software": "Tools used for 16S rRNA gene extraction. Add names and versions of software(s), parameters used",
    "16s_recovered": "Can a 16S gene be recovered from the submitted bin, SAG or MAG?",
    "HRT": "whether subject had hormone replacement theraphy, and if yes start date",
    "IHMC_medication_code": "can include multiple medication codes",
    "MAG_coverage_software": "Tool(s) used to determine the genome coverage if coverage is used as a binning parameter in the extraction of genomes from metagenomic datasets e.g. bwa, bbmap, bowtie, other",
    "adapters": "Adapters provide priming sequences for both amplification and sequencing of the sample-library fragments. Both adapters should be reported; in uppercase letters",
    "air_particulate_matter_concentration": "Concentration of substances that remain suspended in the air, and comprise mixtures of organic and inorganic substances (PM10 and PM2.5); can report multiple PM's by entering numeric values preceded by name of PM",
    "air_temperature_regimen": "information about treatment involving an exposure to varying temperatures; should include the temperature, treatment duration, interval and total experimental duration; can include different temperature regimens",
    "alkalinity": "alkalinity, the ability of a solution to neutralize acids to the equivalence point of carbonate or bicarbonate",
    "alkyl_diethers": "concentration of alkyl diethers",
    "altitude": "The altitude of the sample is the vertical distance between Earth's surface above Sea Level and the sampled position in the air.",
    "aminopeptidase_activity": "measurement of aminopeptidase activity",
    "ammonium": "concentration of ammonium",
    "amniotic_fluid/color": "specification of the color of the amniotic fluid sample",
    "amniotic_fluid/foetal_health_status": "specification of foetal health status, should also include abortion",
    "amniotic_fluid/gestation_state": "specification of the gestation state",
    "amniotic_fluid/maternal_health_status": "specification of the maternal health status",
    "amount_or_size_of_sample_collected": "The total amount or size (volume (ml), mass (g) or area (m2) ) of sample collected.",
    "annotation_source": "For cases where annotation was provided by a community jamboree or model organism database rather than by a specific submitter",
    "antibiotic_regimen": "information about treatment involving antibiotic administration; should include the name of antibiotic, amount administered, treatment duration, interval and total experimental duration; can include multiple antibiotic regimens",
    "assembly_quality": "The assembly quality category is based on sets of criteria outlined for each assembly quality category. For MISAG/MIMAG; Finished: Single, validated, contiguous sequence per replicon without gaps or ambiguities with a consensus error rate equivalent to Q50 or better. High Quality Draft:Multiple fragments where gaps span repetitive regions. Presence of the 23S, 16S and 5S rRNA genes and at least 18 tRNAs. Medium Quality Draft:Many fragments with little to no review of assembly other than reporting of standard assembly statistics. Low Quality Draft:Many fragments with little to no review of assembly other than reporting of standard assembly statistics. Assembly statistics include, but are not limited to total assembly size, number of contigs, contig N50/L50, and maximum contig length. For MIUVIG; Finished: Single, validated, contiguous sequence per replicon without gaps or ambiguities, with extensive manual review and editing to annotate putative gene functions and transcriptional units. High-quality draft genome: One or multiple fragments, totaling ≥ 90% of the expected genome or replicon sequence or predicted complete. Genome fragment(s): One or multiple fragments, totalling < 90% of the expected genome or replicon sequence, or for which no genome size could be estimated.",
    "assembly_software": "Tool(s) used for assembly, including version number and parameters in the format {software};{version};{parameters} e.g. metaSPAdes;3.11.0;kmer set 21,33,55,77,99,121, default parameters otherwise",
    "atmospheric_data": "measurement of atmospheric data; can include multiple data",
    "bacterial_carbon_production": "measurement of bacterial carbon production",
    "bacterial_production": "bacterial production in the water column measured by isotope uptake",
    "bacterial_respiration": "measurement of bacterial respiration in the water column",
    "barometric_pressure": "force per unit area exerted against a surface by the weight of air above that surface",
    "binning_parameters": "The parameters that have been applied during the extraction of genomes from metagenomic datasets e.g. coverage and kmer",
    "binning_software": "Tool(s) used for the extraction of genomes from metagenomic datasets, where possible include a product ID (PID) of the tool(s) used. E.g. MetaCluster-TA (RRID:SCR_004599) or MaxBin (biotools:maxbin)",
    "bio_material": "Unique identifier that references the biological material from which the sample was obtained and that ideally exists in a curated collection (e.g. stock centres, seed banks, DNA banks). The ID should have the following structure: name of the institution (institution code) followed by the collection code (if available) and the voucher id (institution_code:collection_code:voucher_id). Please note institution codes and collection codes are taken from a controlled vocabulary maintained by the INSDC: https://ftp.ncbi.nih.gov/pub/taxonomy/biocollections/",
    "biochemical_oxygen_demand": "a measure of the relative oxygen-depletion effect of a waste contaminant",
    "biomass": "amount of biomass; should include the name for the part of biomass measured, e.g. microbial, total. can include multiple measurements",
    "birth_control": "specification of birth control medication used",
    "bishomohopanol": "concentration of bishomohopanol",
    "blood/blood_disorder": "History of blood disorders; can include multiple disorders. The terms should be chosen from the DO (Human Disease Ontology) at http://www.disease-ontology.org, hematopoietic system disease (https://disease-ontology.org/?id=DOID:74).",
    "broad-scale_environmental_context": "Report the major environmental system the sample or specimen came from. The system(s) identified should have a coarse spatial grain, to provide the general environmental context of where the sampling was done (e.g. in the desert or a rainforest). We recommend using subclasses of EnvO’s biome class: http://purl.obolibrary.org/obo/ENVO_00000428. EnvO documentation about how to use the field: https://github.com/EnvironmentOntology/envo/wiki/Using-ENVO-with-MIxS.",
    "bromide": "concentration of bromide",
    "calcium": "concentration of calcium",
    "carbon/nitrogen_ratio": "ratio of amount or concentrations of carbon to nitrogen",
    "carbon_dioxide": "carbon dioxide (gas) amount or concentration at the time of sampling",
    "carbon_monoxide": "carbon monoxide (gas) amount or concentration at the time of sampling",
    "cell_line": "cell line from which the sample was obtained",
    "cell_type": "cell type from which the sample was obtained",
    "chemical_administration": "list of chemical compounds administered to the host or site where sampling occurred, and when (e.g. antibiotics, N fertilizer, air filter); can include multiple compounds. For Chemical Entities of Biological Interest ontology (CHEBI) (v111), please see http://purl.bioontology.org/ontology/CHEBI",
    "chemical_mutagen": "treatment involving use of mutagens; should include the name of mutagen, amount administered, treatment duration, interval and total experimental duration; can include multiple mutagen regimens",
    "chemical_oxygen_demand": "a measure of the relative oxygen-depletion effect of a waste contaminant",
    "chimera_check_software": "Tool(s) used for chimera checking, including version number and parameters, to discover and remove chimeric sequences. A chimeric sequence is comprised of two or more phylogenetically distinct parent sequences.",
    "chloride": "concentration of chloride",
    "chlorophyll": "concentration of chlorophyll",
    "climate_environment": "treatment involving an exposure to a particular climate; can include multiple climates",
    "collected_by": "name of persons or institute who collected the specimen",
    "collection_date": "The date the sample was collected with the intention of sequencing, either as an instance (single point in time) or interval. In case no exact time is available, the date/time can be right truncated i.e. all of these are valid ISO8601 compliant times: 2008-01-23T19:23:10+00:00; 2008-01-23T19:23:10; 2008-01-23; 2008-01; 2008.",
    "completeness_approach": "The approach used to determine the completeness of a given bin, SAG or MAG, which would typically make use of a set of conserved marker genes or a closely related reference genome. For UViG completeness, include reference genome or group used, and contig feature suggesting a complete genome",
    "completeness_score": "Completeness score is typically based on either the fraction of markers found as compared to a database or the percent of a genome found as compared to a closely related reference genome. Completeness score is one of 3 attributes which in combination reflect the standard quality of a MAG, see here for more information: https://ena-docs.readthedocs.io/en/latest/faq_metagenomes.html. Mandatory for all samples directly linked with SAGs or MAGs.",
    "completeness_software": "Tools used for completion estimate, i.e. checkm, anvi'o, busco. Mandatory for all samples directly linked with SAGs or MAGs.",
    "composite_design/sieving_(if_any)": "collection design of pooled samples and/or sieve size and amount of sample sieved",
    "conductivity": "electrical conductivity of water",
    "contamination_score": "The contamination score is based on the fraction of single-copy genes that are observed more than once in a query genome. Contamination score is one of 3 attributes which in combination reflect the standard quality of a MAG, see here for more information: https://ena-docs.readthedocs.io/en/latest/faq_metagenomes.html. If contamination ≥ 10% then please submit as a binned assembly. In SAGs, contamination score is mandatory for prokaryotes and optional for eukaryotes.",
    "contamination_screening_input": "The type of sequence data used as input",
    "contamination_screening_parameters": "Specific parameters used in the decontamination sofware, such as reference database, coverage, and kmers. Combinations of these parameters may also be used, i.e. kmer and coverage, or reference database and kmer",
    "cultivar": "cultivar (cultivated variety) of plant from which sample was obtained",
    "culture_collection": "Unique identifier that references the culture (e.g. live microbial and viral cultures and cell lines) from which the sample has been obtained and that have been deposited in curated culture collections. The ID needs to provide an institution code and the culture id, with optional collection code, in the following structure: (-institution_code:(collection_code):voucher_id. Please note institution codes (and optional collection codes) are taken from a controlled vocabulary maintained by the INSDC: https://ftp.ncbi.nih.gov/pub/taxonomy/biocollections/",
    "current_land_use": "present state of sample site",
    "current_vegetation": "vegetation classification from one or more standard classification systems, or agricultural crop",
    "current_vegetation_method": "reference or method used in vegetation classification",
    "decontamination_software": "Tool(s) used in contamination screening",
    "density": "density of sample",
    "depth": "The vertical distance below local surface, e.g. for sediment or soil samples depth is measured from sediment or soil surface, respectively. Depth can be reported as an interval for subsurface samples.",
    "dermatology_disorder": "History of dermatology disorders; can include multiple disorders. The terms should be chosen from the DO (Human Disease Ontology) at http://www.disease-ontology.org, skin disease (https://disease-ontology.org/?id=DOID:37).",
    "dev_stage": "if the sample was obtained from an organism in a specific developmental stage, it is specified with this qualifier",
    "diether_lipids": "concentration of diether lipids; can include multiple types of diether lipids",
    "dissolved_carbon_dioxide": "concentration of dissolved carbon dioxide",
    "dissolved_hydrogen": "concentration of dissolved hydrogen",
    "dissolved_inorganic_carbon": "dissolved inorganic carbon concentration",
    "dissolved_inorganic_nitrogen": "concentration of dissolved inorganic nitrogen",
    "dissolved_inorganic_phosphorus": "concentration of dissolved inorganic phosphorus",
    "dissolved_organic_carbon": "concentration of dissolved organic carbon",
    "dissolved_organic_nitrogen": "dissolved organic nitrogen concentration measured as; total dissolved nitrogen - NH4 - NO3 - NO2",
    "dissolved_oxygen": "concentration of dissolved oxygen",
    "dominant_hand": "dominant hand of the subject",
    "douche": "date of most recent douche",
    "downward_PAR": "visible waveband radiance and irradiance measurements in the water column",
    "drainage_classification": "drainage classification from a standard system such as the USDA system",
    "drug_usage": "any drug used by subject and the frequency of usage; can include multiple drugs used",
    "ecotype": "a population within a given species displaying genetically based, phenotypic traits that reflect adaptation to a local habitat.",
    "efficiency_percent": "percentage of volatile solids removed from the anaerobic digestor",
    "elevation": "The elevation of the sampling site as measured by the vertical distance from mean sea level.",
    "emulsions": "amount or concentration of substances such as paints, adhesives, mayonnaise, hair colorants, emulsified oils, etc.; can include multiple emulsion types",
    "encoded_traits": "Should include key traits like antibiotic resistance or xenobiotic degradation phenotypes for plasmids, converting genes for phage",
    "environmental_medium": "Report the environmental material(s) immediately surrounding the sample or specimen at the time of sampling. We recommend using subclasses of 'environmental material' (http://purl.obolibrary.org/obo/ENVO_00010483). EnvO documentation about how to use the field: https://github.com/EnvironmentOntology/envo/wiki/Using-ENVO-with-MIxS . Terms from other OBO ontologies are permissible as long as they reference mass/volume nouns (e.g. air, water, blood) and not discrete, countable entities (e.g. a tree, a leaf, a table top).",
    "environmental_sample": "identifies sequences derived by direct molecular isolation from a bulk environmental DNA sample (by PCR with or without subsequent cloning of the product, DGGE, or other anonymous methods) with no reliable identification of the source organism",
    "estimated_size": "The estimated size of the genome (in bp) prior to sequencing. Of particular importance in the sequencing of (eukaryotic) genome which could remain in draft form for a long or unspecified period. Mandatory for MIGS of eukaryotes.",
    "ethnicity": "A category of people who identify with each other, usually on the basis of presumed similarities such as a common language, ancestry, history, society, culture, nation or social treatment within their residing area. https://en.wikipedia.org/wiki/List_of_contemporary_ethnic_groups",
    "experimental_factor": "Experimental factors are essentially the variable aspects of an experiment design which can be used to describe an experiment, or set of experiments, in an increasingly detailed manner. This field accepts ontology terms from Experimental Factor Ontology (EFO) and/or Ontology for Biomedical Investigations (OBI). For a browser of EFO (v 2.95) terms, please see http://purl.bioontology.org/ontology/EFO; for a browser of OBI (v 2018-02-12) terms please see http://purl.bioontology.org/ontology/OBI. E.g. time series design [EFO:EFO_0001779]",
    "extrachromosomal_elements": "Do plasmids exist of significant phenotypic consequence (e.g. ones that determine virulence or antibiotic resistance). Megaplasmids? Other plasmids (borrelia has 15+ plasmids).",
    "extreme_unusual_properties/Al_saturation": "aluminum saturation (esp. for tropical soils)",
    "extreme_unusual_properties/Al_saturation_method": "reference or method used in determining Al saturation",
    "extreme_unusual_properties/heavy_metals": "heavy metals present and concentrations of any drug used by subject and the frequency of usage; can include multiple heavy metals and concentrations",
    "extreme_unusual_properties/heavy_metals_method": "reference or method used in determining heavy metals",
    "feature_prediction": "Method used to predict UViGs features such as ORFs, integration site, etc. Add names and versions of software(s), parameters used",
    "fertilizer_regimen": "information about treatment involving the use of fertilizers; should include the name fertilizer, amount administered, treatment duration, interval and total experimental duration; can include multiple fertilizer regimens",
    "fluorescence": "raw (volts) or converted (mg Chla/m^3) fluorescence of the water",
    "fungicide_regimen": "information about treatment involving use of fungicides; should include the name of fungicide, amount administered, treatment duration, interval and total experimental duration; can include multiple fungicide regimens",
    "gaseous_environment": "use of conditions with differing gaseous environments; should include the name of gaseous compound, amount administered, treatment duration, interval and total experimental duration; can include multiple gaseous environment regimens",
    "gaseous_substances": "amount or concentration of substances such as hydrogen sulfide, carbon dioxide, methane, etc.; can include multiple substances",
    "gastrointestinal_tract_disorder": "History of gastrointestinal tract disorders; can include multiple disorders. History of blood disorders; can include multiple disorders. The terms should be chosen from the DO (Human Disease Ontology) at http://www.disease-ontology.org, gastrointestinal system disease (https://disease-ontology.org/?id=DOID:77).",
    "geographic_location_(country_and/or_sea)": "The geographical origin of where the sample was collected from, with the intention of sequencing, as defined by the country or sea name. Country or sea names should be chosen from the INSDC country list (http://insdc.org/country.html).",
    "geographic_location_(latitude)": "The geographical origin of the sample as defined by latitude. The values should be reported in decimal degrees and in WGS84 system",
    "geographic_location_(longitude)": "The geographical origin of the sample as defined by longitude. The values should be reported in decimal degrees and in WGS84 system",
    "geographic_location_(region_and_locality)": "The geographical origin of the sample as defined by the specific region name followed by the locality name.",
    "germline": "the sample described presented in the entry has not undergone somatic genomic rearrangement as part of an adaptive immune response; it is the unrearranged molecule that was inherited from the parental germline",
    "glucosidase_activity": "measurement of glucosidase activity",
    "gravidity": "Whether or not the subject is gravid. If so, report date due or date post-conception and specify which of these two dates is being reported.",
    "gravity": "information about treatment involving use of gravity factor to study various types of responses in presence, absence or modified levels of gravity; can include multiple treatments",
    "growth_hormone_regimen": "information about treatment involving use of growth hormones; should include the name of growth hormone, amount administered, treatment duration, interval and total experimental duration; can include multiple growth hormone regimens",
    "growth_media": "information about growth media for growing the plants or tissue cultured samples",
    "gynecological_disorder": "History of gynecological disorders; can include multiple disorders. The terms should be chosen from the DO (Human Disease Ontology) at http://www.disease-ontology.org, female reproductive system disease (https://disease-ontology.org/?id=DOID:229).",
    "herbicide_regimen": "information about treatment involving use of herbicides; information about treatment involving use of growth hormones; should include the name of herbicide, amount administered, treatment duration, interval and total experimental duration; can include multiple regimens",
    "history/agrochemical_additions": "addition of fertilizers, pesticides, etc. - amount and time of applications",
    "history/crop_rotation": "whether or not crop is rotated, and if yes, rotation schedule",
    "history/extreme_events": "unusual physical events that may have affected microbial populations",
    "history/fire": "historical and/or physical evidence of fire",
    "history/flooding": "historical and/or physical evidence of flooding",
    "history/previous_land_use": "previous land use and dates",
    "history/previous_land_use_method": "reference or method used in determining previous land use and dates",
    "history/tillage": "note method(s) used for tilling",
    "host_HIV_status": "HIV status of subject, if yes HAART initiation status should also be indicated as [YES or NO]",
    "host_age": "age of host at the time of sampling; relevant scale depends on species and study, e.g. could be seconds for amoebae or centuries for trees",
    "host_blood_pressure_diastolic": "resting diastolic blood pressure, measured as mm mercury",
    "host_blood_pressure_systolic": "resting systolic blood pressure, measured as mm mercury",
    "host_body-mass_index": "body mass index, calculated as weight/(height)squared",
    "host_body_habitat": "original body habitat where the sample was obtained from",
    "host_body_product": "substance produced by the body, e.g. stool, mucus, where the sample was obtained from. For Foundational Model of Anatomy Ontology (FMA) (v 3.1) terms, please see http://purl.bioontology.org/ontology/FMA",
    "host_body_site": "name of body site where the sample was obtained from, such as a specific organ or tissue (tongue, lung etc...). For Foundational Model of Anatomy Ontology (FMA) (v 3.1) terms, please see http://purl.bioontology.org/ontology/FMA",
    "host_body_temperature": "core body temperature of the host when sample was collected",
    "host_color": "the color of host",
    "host_common_name": "common name of the host, e.g. human",
    "host_diet": "type of diet depending on the host, for animals omnivore, herbivore etc., for humans high-fat, Mediterranean etc.; can include multiple diet types",
    "host_disease_status": "list of diseases with which the host has been diagnosed; can include multiple diagnoses. The value of the field depends on host; for humans the terms should be chosen from DO (Disease Ontology) at http://www.disease-ontology.org, other hosts are free text",
    "host_dry_mass": "measurement of dry mass",
    "host_family_relationship": "relationships to other hosts in the same study; can include multiple relationships",
    "host_genotype": "observed genotype",
    "host_growth_conditions": "literature reference giving growth conditions of the host",
    "host_height": "the height of subject",
    "host_last_meal": "content of last meal and time since feeding; can include multiple values",
    "host_length": "the length of subject",
    "host_life_stage": "description of life stage of host",
    "host_occupation": "most frequent job performed by subject",
    "host_phenotype": "phenotype of host. For Phenotypic quality Ontology (PATO) (v 2013-10-28) terms, please see http://purl.bioontology.org/ontology/PATO",
    "host_pulse": "resting pulse, measured as beats per minute",
    "host_scientific_name": "Scientific name of the natural (as opposed to laboratory) host to the organism from which sample was obtained.",
    "host_sex": "Gender or sex of the host.",
    "host_shape": "morphological shape of host",
    "host_subject_id": "a unique identifier by which each subject can be referred to, de-identified, e.g. #131",
    "host_subspecific_genetic_lineage": "Information about the genetic distinctness of the host organism below the subspecies level e.g., serovar, serotype, biotype, ecotype, variety, cultivar, or any relevant genetic typing schemes like Group I plasmid. Subspecies should not be recorded in this term, but in the NCBI taxonomy. Supply both the lineage name and the lineage rank separated by a colon, e.g., biovar:abc123.",
    "host_substrate": "the growth substrate of the host",
    "host_taxid": "NCBI taxon id of the host, e.g. 9606",
    "host_total_mass": "total mass of the host at collection, the unit depends on host",
    "host_wet_mass": "measurement of wet mass",
    "humidity": "amount of water vapour in the air, at the time of sampling",
    "humidity_regimen": "information about treatment involving an exposure to varying degree of humidity; information about treatment involving use of growth hormones; should include amount of humidity administered, treatment duration, interval and total experimental duration; can include multiple regimens",
    "hysterectomy": "specification of whether hysterectomy was performed",
    "identified_by": "name of the expert who identified the specimen taxonomically",
    "industrial_effluent_percent": "percentage of industrial effluents received by wastewater treatment plant",
    "inorganic_particles": "concentration of particles such as sand, grit, metal particles, ceramics, etc.; can include multiple particles",
    "investigation_type": "Nucleic Acid Sequence Report is the root element of all MIxS compliant reports as standardised by Genomic Standards Consortium",
    "isolate": "individual isolate from which the sample was obtained",
    "isolation_and_growth_condition": "Publication reference in the form of pubmed ID (pmid), digital object identifier (doi) or url for isolation and growth condition specifications of the organism/material. Mandatory for MIGS and MIMARKS Specimen.",
    "isolation_source": "describes the physical, environmental and/or local geographical source of the biological sample from which the sample was derived",
    "known_pathogenicity": "To what is the entity pathogenic, for instance plant, fungi, bacteria",
    "lab_host": "scientific name of the laboratory host used to propagate the source organism from which the sample was obtained",
    "lat_lon": "geographical coordinates of the location where the specimen was collected",
    "library_construction_method": "Library construction method used for clone libraries",
    "library_reads_sequenced": "Total number of clones sequenced from the library",
    "library_screening_strategy": "Specific enrichment or screening methods applied before and/or after creating clone libraries in order to select a specific group of sequences",
    "library_size": "Total number of clones in the library prepared for the project",
    "library_vector": "Cloning vector type(s) used in construction of libraries",
    "light_intensity": "measurement of light intensity",
    "link_to_classification_information": "link to digitized soil maps or other soil classification information",
    "link_to_climate_information": "link to climate resource",
    "links_to_additional_analysis": "link to additional analysis results performed on the sample",
    "liver_disorder": "History of liver disorders; can include multiple disorders. The terms should be chosen from the DO (Human Disease Ontology) at http://www.disease-ontology.org, liver disease (https://disease-ontology.org/?id=DOID:409).",
    "local_environmental_context": "Report the entity or entities which are in the sample or specimen’s local vicinity and which you believe have significant causal influences on your sample or specimen. We recommend using EnvO terms which are of smaller spatial grain than your entry for \"broad-scale environmental context\". Terms, such as anatomical sites, from other OBO Library ontologies which interoperate with EnvO (e.g. UBERON) are accepted in this field. EnvO documentation about how to use the field: https://github.com/EnvironmentOntology/envo/wiki/Using-ENVO-with-MIxS.",
    "lung/nose-throat_disorder": "History of nose-throat disorders; can include multiple disorders, The terms should be chosen from the DO (Human Disease Ontology) at http://www.disease-ontology.org, lung disease (https://disease-ontology.org/?id=DOID:850), upper respiratory tract disease (https://disease-ontology.org/?id=DOID:974).",
    "lung/pulmonary_disorder": "History of pulmonary disorders; can include multiple disorders. The terms should be chosen from the DO (Human Disease Ontology) at http://www.disease-ontology.org, lung disease (https://disease-ontology.org/?id=DOID:850).",
    "magnesium": "concentration of magnesium",
    "major_diet_change_in_last_six_months": "specification of major diet changes in the last six months, if yes the change should be specified",
    "mating_type": "mating type of the organism from which the sequence was obtained; mating type is used for prokaryotes, and for eukaryotes that undergo meiosis without sexually dimorphic gametes",
    "mean_annual_and_seasonal_precipitation": "mean annual and seasonal precipitation (mm)",
    "mean_annual_and_seasonal_temperature": "mean annual and seasonal temperature (ºC)",
    "mean_friction_velocity": "measurement of mean friction velocity",
    "mean_peak_friction_velocity": "measurement of mean peak friction velocity",
    "mechanical_damage": "information about any mechanical damage exerted on the plant; can include multiple damages and sites",
    "medical_history_performed": "whether full medical history was collected",
    "menarche": "date of most recent menstruation",
    "menopause": "date of onset of menopause",
    "metagenomic_source": "The metagenomic source of the sample. This value should contain “metagenome” and be in the taxonomy database e.g. wastewater metagenome or human gut metagenome. Please note “metagenome” alone will not be accepted. Check here for more details on metagenome taxonomy: https://ena-docs.readthedocs.io/en/latest/faq_taxonomy.html#environmental-taxonomic-classifications",
    "methane": "methane (gas) amount or concentration at the time of sampling",
    "microbial_biomass": "the part of the organic matter in the soil that constitutes living microorganisms smaller than 5-10 µm. IF you keep this, you would need to have correction factors used for conversion to the final units, which should be mg C (or N)/kg soil).",
    "microbial_biomass_method": "reference or method used in determining microbial biomass",
    "mineral_nutrient_regimen": "information about treatment involving the use of mineral supplements; should include the name of mineral nutrient, amount administered, treatment duration, interval and total experimental duration; can include multiple mineral nutrient regimens",
    "multiplex_identifiers": "Molecular barcodes, called Multiplex Identifiers (MIDs), that are used to specifically tag unique samples in a sequencing run. Sequence should be reported in uppercase letters",
    "n-alkanes": "concentration of n-alkanes; can include multiple n-alkanes",
    "negative_control_type": "The substance or equipment used as a negative control in an investigation",
    "nitrate": "concentration of nitrate",
    "nitrite": "concentration of nitrite",
    "nitrogen": "concentration of nitrogen (total)",
    "non-mineral_nutrient_regimen": "information about treatment involving the exposure of plant to non-mineral nutrient such as oxygen, hydrogen or carbon; should include the name of non-mineral nutrient, amount administered, treatment duration, interval and total experimental duration; can include multiple non-mineral nutrient regimens",
    "nose/mouth/teeth/throat_disorder": "History of nose/mouth/teeth/throat disorders; can include multiple disorders. The terms should be chosen from the DO (Human Disease Ontology) at http://www.disease-ontology.org, nose disease (https://disease-ontology.org/?id=DOID:2825), mouth disease (https://disease-ontology.org/?id=DOID:403), tooth disease (https://disease-ontology.org/?id=DOID:1091), or upper respiratory tract disease (https://disease-ontology.org/?id=DOID:974).",
    "nucleic_acid_amplification": "A link to a literature reference, electronic resource or a standard operating procedure (SOP), that describes the enzymatic amplification (PCR, TMA, NASBA) of specific nucleic acids",
    "nucleic_acid_extraction": "A link to a literature reference, electronic resource or a standard operating procedure (SOP), that describes the material separation to recover the nucleic acid fraction from a sample",
    "number_of_replicons": "Reports the number of replicons in a nuclear genome of eukaryotes, in the genome of a bacterium or archaea or the number of segments in a segmented virus. Always applied to the haploid chromosome count of a eukaryote. Mandatory for MIGS of eukaryotes, bacteria, archaea and segmented virus.",
    "number_of_standard_tRNAs_extracted": "The total number of tRNAs identified from the bin, SAG or MAG",
    "observed_biotic_relationship": "Is it free-living or in a host and if the latter what type of relationship is observed",
    "organic_carbon": "concentration of organic carbon",
    "organic_matter": "concentration of organic matter",
    "organic_nitrogen": "concentration of organic nitrogen",
    "organic_particles": "concentration of particles such as faeces, hairs, food, vomit, paper fibers, plant material, humus, etc.",
    "organism_count": "Total cell count of any organism (or group of organisms) per gram, volume or area of sample, should include name of organism followed by count. The method that was used for the enumeration (e.g. qPCR, atp, mpn, etc.) Should also be provided. (example: total prokaryotes; 3.5e7 cells per ml; qpcr)",
    "oxygen": "oxygen (gas) amount or concentration at the time of sampling",
    "oxygenation_status_of_sample": "oxygenation status of sample",
    "pH": "pH measurement",
    "pH_method": "reference or method used in determining pH",
    "pH_regimen": "information about treatment involving exposure of plants to varying levels of pH of the growth media; can include multiple regimen",
    "particulate_organic_carbon": "concentration of particulate organic carbon",
    "particulate_organic_nitrogen": "concentration of particulate organic nitrogen",
    "pcr_conditions": "Description of reaction conditions and components for PCR in the form of 'initial denaturation:94degC_1.5min; annealing=...'",
    "pcr_primers": "PCR primers that were used to amplify the sequence of the targeted gene, locus or subfragment. This field should contain all the primers used for a single PCR reaction if multiple forward or reverse primers are present in a single PCR reaction. The primer sequence should be reported in uppercase letters",
    "perturbation": "type of perturbation, e.g. chemical administration, physical disturbance, etc., coupled with time that perturbation occurred; can include multiple perturbation types",
    "pesticide_regimen": "information about treatment involving use of insecticides; should include the name of pesticide, amount administered, treatment duration, interval and total experimental duration; can include multiple pesticide regimens",
    "petroleum_hydrocarbon": "concentration of petroleum hydrocarbon",
    "phaeopigments": "concentration of phaeopigments; can include multiple phaeopigments",
    "phosphate": "concentration of phosphate",
    "phospholipid_fatty_acid": "concentration of phospholipid fatty acids; can include multiple values",
    "photon_flux": "measurement of photon flux",
    "plant_body_site": "name of body site that the sample was obtained from. For Plant Ontology (PO) (v 20) terms, see http://purl.bioontology.org/ontology/PO",
    "plant_product": "substance produced by the plant, where the sample was obtained from",
    "ploidy": "The ploidy level of the genome (e.g. allopolyploid, haploid, diploid, triploid, tetraploid). It has implications for the downstream study of duplicated gene and regions of the genomes (and perhaps for difficulties in assembly). For terms, please select terms listed under class ploidy (PATO:001374) of Phenotypic Quality Ontology (PATO), and for a browser of PATO (v 2018-03-27) please refer to http://purl.bioontology.org/ontology/PATO",
    "pollutants": "pollutant types and, amount or concentrations measured at the time of sampling; can report multiple pollutants by entering numeric values preceded by name of pollutant",
    "pooling_of_DNA_extracts_(if_done)": "were multiple DNA extractions mixed? how many?",
    "positive_control_type": "The substance, mixture, product, or apparatus used to verify that a process which is part of an investigation delivers a true positive.",
    "potassium": "concentration of potassium",
    "pre-treatment": "the process of pre-treatment removes materials that can be easily collected from the raw wastewater",
    "pregnancy": "date due of pregnancy",
    "presence_of_pets_or_farm_animals": "specification of presence of pets or farm animals in the environment of subject, if yes the animals should be specified; can include multiple animals present",
    "pressure": "pressure to which the sample is subject, in atmospheres",
    "primary_production": "measurement of primary production",
    "primary_treatment": "the process to produce both a generally homogeneous liquid capable of being treated biologically and a sludge that can be separately treated or processed",
    "profile_position": "cross-sectional position in the hillslope where sample was collected.sample area position in relation to surrounding areas",
    "project_name": "Name of the project within which the sequencing was organized",
    "propagation": "The type of reproduction from the parent stock. Values for this field is specific to different taxa. For phage or virus: lytic/lysogenic/temperate/obligately lytic. For plasmids: incompatibility group. For eukaryotes: sexual/asexual. Mandatory for MIGs of eukayotes, plasmids and viruses.",
    "radiation_regimen": "information about treatment involving exposure of plant or a plant part to a particular radiation regimen; should include the radiation type, amount or intensity administered, treatment duration, interval and total experimental duration; can include multiple radiation regimens",
    "rainfall_regimen": "information about treatment involving an exposure to a given amount of rainfall; can include multiple regimens",
    "reactor_type": "anaerobic digesters can be designed and engineered to operate using a number of different process configurations, as batch or continuous, mesophilic, high solid or low solid, and single stage or multistage",
    "reassembly_post_binning": "Has an assembly been performed on a genome bin extracted from a metagenomic assembly?",
    "redox_potential": "redox potential, measured relative to a hydrogen cell, indicating oxidation or reduction potential",
    "reference_database(s)": "List of database(s) used for ORF annotation, along with version number and reference to website or publication",
    "reference_for_biomaterial": "Primary publication if isolated before genome publication; otherwise, primary genome report. Mandatory for MIGS of bacteria and archaea.",
    "relationship_to_oxygen": "Is this organism an aerobe, anaerobe? Please note that aerobic and anaerobic are valid descriptors for microbial environments",
    "relevant_electronic_resources": "A related resource that is referenced, cited, or otherwise associated to the sequence in the format of a PMID, DOI or URL",
    "relevant_standard_operating_procedures": "Standard operating procedures used in assembly and/or annotation of genomes, metagenomes or environmental sequences in the format of a PMID, DOI or URL",
    "salinity": "The total concentration of all dissolved salts in a liquid or solid sample. While salinity can be measured by a complete chemical analysis, this method is difficult and time consuming. More often, it is instead derived from the conductivity measurement. This is known as practical salinity. These derivations compare the specific conductance of the sample to a salinity standard such as seawater.",
    "salinity_method": "reference or method used in determining salinity",
    "salt_regimen": "information about treatment involving use of salts as supplement to liquid and soil growth media; should include the name of salt, amount administered, treatment duration, interval and total experimental duration; can include multiple salt regimens",
    "sample_collection_device": "The device used to collect an environmental sample. It is recommended to use terms listed under environmental sampling device (http://purl.obolibrary.org/obo/ENVO) and/or terms listed under specimen collection device (http://purl.obolibrary.org/obo/GENEPIO_0002094).",
    "sample_collection_method": "The method employed for collecting the sample. Can be provided in the form of a PMID, DOI, url or text.",
    "sample_derived_from": "Reference to parental sample(s) or original run(s) that the assembly is derived from. The referenced samples or runs should already be registered in INSDC. This should be formatted as one of the following. A single sample/run e.g. ERSxxxxxx OR a comma separated list e.g. ERSxxxxxx,ERSxxxxxx OR a range e.g. ERSxxxxxx-ERSxxxxxx",
    "sample_material_processing": "A brief description of any processing applied to the sample during or after retrieving the sample from environment, or a link to the relevant protocol(s) performed.",
    "sample_storage_duration": "duration for which sample was stored",
    "sample_storage_location": "location at which sample was stored, usually name of a specific freezer/room",
    "sample_storage_temperature": "temperature at which sample was stored, e.g. -80",
    "sample_volume_or_weight_for_DNA_extraction": "Volume (ml) or mass (g) of total collected sample processed for DNA extraction. Note: total sample collected should be entered under the term 'sample size'.",
    "sample_weight_for_DNA_extraction": "weight (g) of soil processed",
    "seasonal_environment": "treatment involving an exposure to a particular season (e.g. winter, summer, rabi, rainy etc.)",
    "secondary_treatment": "the process for substantially degrading the biological content of the sewage",
    "sequence_quality_check": "Indicate if the sequence has been called by automatic systems (none) or undergone a manual editing procedure (e.g. by inspecting the raw data or chromatograms). Applied only for sequences that are not submitted to SRA or DRA",
    "sequencing_method": "Sequencing machine used. Where possible the term should be taken from the OBI list of DNA sequencers (http://purl.obolibrary.org/obo/OBI_0400103).",
    "serotype": "serological variety of a species characterized by its antigenic properties",
    "serovar": "serological variety of a species (usually a prokaryote) characterized by its antigenic properties",
    "sewage_type": "Type of sewage based on origin: wastewater treatment plant (municipal or industrial), open sewer line, river, stream, stagnant pool, or other.",
    "sex": "sex of the organism from which the sample was obtained",
    "sexual_activity": "current sexual partner and frequency of sex",
    "silicate": "concentration of silicate",
    "similarity_search_method": "Tool used to compare ORFs with database, along with version and cutoffs used. Add names and versions of software(s), parameters used",
    "size_fraction_selected": "Filtering pore size used in sample preparation e.g. 0-0.22 micrometer",
    "slope_aspect": "the direction a slope faces. While looking down a slope use a compass to record the direction you are facing (direction or degrees); e.g., NW or 315°. This measure provides an indication of sun and wind exposure that will influence soil temperature and evapotranspiration.",
    "slope_gradient": "commonly called slope. The angle between ground surface and a horizontal line (in percent). This is the direction that overland water would flow. This measure is usually taken with a hand level meter or clinometer.",
    "sludge_retention_time": "the time activated sludge remains in reactor",
    "smoker": "specification of smoking status",
    "sodium": "sodium concentration",
    "soil_horizon": "specific layer in the land area which measures parallel to the soil surface and possesses physical characteristics which differ from the layers above and beneath",
    "soil_horizon_method": "reference or method used in determining the horizon",
    "soil_taxonomic/FAO_classification": "soil classification from the FAO World Reference Database for Soil Resources",
    "soil_taxonomic/local_classification": "soil classification based on local soil classification system",
    "soil_taxonomic/local_classification_method": "reference or method used in determining the local soil classification",
    "soil_texture_measurement": "the relative proportion of different grain sizes of mineral particles in a soil, as described using a standard system; express as % sand (50 um to 2 mm), silt (2 um to 50 um), and clay (<2 um) with textural name (e.g., silty clay loam) optional.",
    "soil_texture_method": "reference or method used in determining soil texture",
    "soil_type": "Description of the soil type or classification. This field accepts terms under soil (http://purl.obolibrary.org/obo/ENVO_00001998). Multiple terms can be separated by pipes.",
    "soil_type_method": "reference or method used in determining soil series name or other lower-level classification",
    "solar_irradiance": "the amount of solar energy that arrives at a specific area of a surface during a specific time interval",
    "soluble_inorganic_material": "concentration of substances such as ammonia, road-salt, sea-salt, cyanide, hydrogen sulfide, thiocyanates, thiosulfates, etc.",
    "soluble_organic_material": "concentration of substances such as urea, fruit sugars, soluble proteins, drugs, pharmaceuticals, etc.",
    "soluble_reactive_phosphorus": "concentration of soluble reactive phosphorus",
    "source_material_identifiers": "A unique identifier assigned to a material sample (as defined by http://rs.tdwg.org/dwc/terms/materialSampleID, and as opposed to a particular digital record of a material sample) used for extracting nucleic acids, and subsequent sequencing. The identifier can refer either to the original material collected or to any derived sub-samples. The INSDC qualifiers /specimen_voucher, /bio_material, or /culture_collection may or may not share the same value as the source_mat_id field. For instance, the /specimen_voucher qualifier and source_mat_id may both contain 'UAM:Herps:14' , referring to both the specimen voucher and sampled tissue with the same identifier. However, the /culture_collection qualifier may refer to a value from an initial culture (e.g. ATCC:11775) while source_mat_id would refer to an identifier from some derived culture from which the nucleic acids were extracted (e.g. xatc123 or ark:/2154/R2).",
    "special_diet": "specification of special diet; can include multiple special diets",
    "specimen_voucher": "Unique identifier that references the physical specimen that remains after the sequence has been obtained and that ideally exists in a curated collection. The ID should have the following structure: name of the institution (institution code) followed by the collection code (if available) and the voucher id (institution_code:collection_code:voucher_id). Please note institution codes and collection codes are taken from a controlled vocabulary maintained by the INSDC: https://ftp.ncbi.nih.gov/pub/taxonomy/biocollections/",
    "standing_water_regimen": "treatment involving an exposure to standing water during a plant's life span, types can be flood water or standing water; can include multiple regimens",
    "storage_conditions_(fresh/frozen/other)": "explain how and for how long the soil sample was stored before DNA extraction.",
    "strain": "Name of the strain from which the sample was obtained.",
    "study_completion_status": "specification of study completion status, if no the reason should be specified",
    "sub_species": "name of sub-species of organism from which sample was obtained",
    "sub_strain": "name or identifier of a genetically or otherwise modified strain from which sample was obtained, derived from a parental strain (which should be annotated in the strain field; sub_strain from which sample was obtained",
    "subspecific_genetic_lineage": "Information about the genetic distinctness of the sequenced organism below the subspecies level, e.g., serovar, serotype, biotype, ecotype, or any relevant genetic typing schemes like Group I plasmid. Subspecies should not be recorded in this term, but in the NCBI taxonomy. Supply both the lineage name and the lineage rank separated by a colon, e.g., biovar:abc123.",
    "sulfate": "concentration of sulfate",
    "sulfide": "concentration of sulfide",
    "suspended_particulate_matter": "concentration of suspended particulate matter",
    "suspended_solids": "concentration of substances including a wide variety of material, such as silt, decaying plant and animal matter, etc,; can include multiple substances",
    "tRNA_extraction_software": "Tools used for tRNA identification. Add names and versions of software(s), parameters used",
    "target_gene": "Targeted gene or locus name for marker gene studies",
    "target_subfragment": "Name of subfragment of a gene or locus. Important to e.g. identify special regions on marker genes like V6 on 16S rRNA",
    "taxonomic_classification": "Method used for taxonomic classification, along with reference database used, classification rank, and thresholds used to classify new genomes. Expected values are: classification method, database name, and other parameters e.g. vConTACT vContact2 (references from NCBI RefSeq v83, genus rank classification, default parameters)",
    "taxonomic_identity_marker": "The phylogenetic marker(s) used to assign an organism name to the bin, SAG or MAG. Examples are 16S gene, multi-marker approach or other e.g. rpoB gene. Mandatory for all samples directly linked with SAGs or MAGs.",
    "temperature": "temperature of the sample at time of sampling",
    "tertiary_treatment": "the process providing a final treatment stage to raise the effluent quality before it is discharged to the receiving environment",
    "tidal_stage": "stage of tide",
    "time_since_last_toothbrushing": "specification of the time since last toothbrushing",
    "time_since_last_wash": "specification of the time since last wash",
    "tissue_culture_growth_media": "description of plant tissue culture growth media used",
    "tissue_lib": "tissue library from which sample was obtained",
    "tissue_type": "tissue type from which the sample was obtained",
    "total_carbon": "total carbon content",
    "total_depth_of_water_column": "measurement of total depth of water column",
    "total_dissolved_nitrogen": "total dissolved nitrogen concentration, reported as nitrogen, measured by: total dissolved nitrogen = NH4 + NO3NO2 + dissolved organic nitrogen",
    "total_inorganic_nitrogen": "total inorganic nitrogen content",
    "total_nitrogen": "total nitrogen content of the sample",
    "total_nitrogen_method": "reference or method used in determining the total nitrogen",
    "total_organic_C_method": "reference or method used in determining total organic C",
    "total_organic_carbon": "Definition for soil: total organic C content of the soil units of g C/kg soil. Definition otherwise: total organic carbon content",
    "total_particulate_carbon": "total particulate carbon content",
    "total_phosphate": "total amount or concentration of phosphate",
    "total_phosphorus": "total phosphorus concentration, calculated by: total phosphorus = total dissolved phosphorus + particulate phosphorus. Can also be measured without filtering, reported as phosphorus",
    "travel_outside_the_country_in_last_six_months": "specification of the countries travelled in the last six months; can include multiple travels",
    "trophic_level": "Trophic levels are the feeding position in a food chain. Microbes can be a range of producers (e.g. chemolithotroph)",
    "turbidity": "turbidity measurement",
    "twin_sibling_presence": "specification of twin sibling presence",
    "urine/collection_method": "specification of urine collection method",
    "urine/kidney_disorder": "History of kidney disorders; can include multiple disorders. The terms should be chosen from the DO (Human Disease Ontology) at http://www.disease-ontology.org, kidney disease (https://disease-ontology.org/?id=DOID:557).",
    "urine/urogenital_tract_disorder": "History of urogenital tract disorders; can include multiple disorders. The terms should be chosen from the DO (Human Disease Ontology) at http://www.disease-ontology.org, urinary system disease (https://disease-ontology.org/?id=DOID:18).",
    "urogenital_disorder": "History of urogenital disorders, can include multiple disorders. The terms should be chosen from the DO (Human Disease Ontology) at http://www.disease-ontology.org, reproductive system disease (https://disease-ontology.org/?id=DOID:15) or urinary system disease (https://disease-ontology.org/?id=DOID:18).",
    "variety": "variety (= varietas, a formal Linnaean rank) of organism from which sample was derived.",
    "ventilation_rate": "ventilation rate of the system in the sampled premises",
    "ventilation_type": "The intentional movement of air from outside a building to the inside through forced or natural movement of air",
    "volatile_organic_compounds": "concentration of carbon-based chemicals that easily evaporate at room temperature; can report multiple volatile organic compounds by entering numeric values preceded by name of compound",
    "wastewater_type": "the origin of wastewater such as human waste, rainfall, storm drains, etc.",
    "water_content": "water content measurement",
    "water_content_method": "reference or method used in determining the water content of soil",
    "water_current": "measurement of magnitude and direction of flow within a fluid",
    "water_temperature_regimen": "information about treatment involving an exposure to water with varying degree of temperature; can include multiple regimens",
    "watering_regimen": "information about treatment involving an exposure to watering frequencies; can include multiple regimens",
    "weight_loss_in_last_three_months": "specification of weight loss in the last three months, if yes should be further specified to include amount of weight loss",
    "wind_direction": "wind direction is the direction from which a wind originates",
    "wind_speed": "speed of wind measured at the time of sampling",
}
